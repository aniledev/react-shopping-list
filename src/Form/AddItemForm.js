import React from "react";

// a form with an input and a submit button to add items to the list

export default class AddItemForm extends React.Component {
  // buttons are not wired up, not functional yet
  render() {
    return (
      <form className="form">
        <label htmlFor="add-item" className="add-form" id="add-form">
          Add Item
        </label>
        <input
          type="text"
          className="add-item"
          id="add-item"
          placeholder="carrots"
          aria-label="Shopping list item"
          min={1}
        ></input>
        <button type="submit">Add Item</button>
      </form>
    );
  }
}
