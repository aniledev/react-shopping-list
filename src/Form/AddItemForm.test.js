import React from "react";
import ReactDOM from "react-dom";
import AddItemForm from "./AddItemForm";

describe("AddItemForm Component", () => {
  // start with a smoke test inside a describe block; does the component render properly without errors
  it("should render without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AddItemForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
