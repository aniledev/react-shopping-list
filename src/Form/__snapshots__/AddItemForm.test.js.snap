// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm Component should render the complete form 1`] = `
<form
  className="form"
>
  <label
    className="add-form"
    htmlFor="add-item"
    id="add-form"
  >
    Add Item
  </label>
  <input
    aria-label="Shopping list item"
    className="add-item"
    id="add-item"
    min={1}
    placeholder="carrots"
    type="text"
  />
  <button
    type="submit"
  >
    Add Item
  </button>
</form>
`;
